### PR TITLE
Propose additions to top-level README and new project structure

### PR DESCRIPTION
**Pull request by evazorro**
Add statement about intent of docs, pull some overview content into this top-level README, then add a section directing readers to different parts of the docs based on what they're trying to accomplish. 

Also include a (fake) table of contents with my proposed restructuring of the documentation. Notes are given in "//" comments and reference existing content in relative file paths where appropriate.

### CODE REVIEW

**Comment by kytrinyx** [comment]
@evazorro Just a quick note to say that this is super high on my list, but that I'm wrestling some bugs in the CLI, which is affecting a bunch of people. I expect to be able to tackle this properly starting September 1, at which point it will be my primary focus (aside from the day-job, at least).

**Comment by mbbroberg** [comment]
Hi @evazorro! I'm helping review this PR while I consult with the Exercism team. I'm really excited to dive in - thank you for making it so easy to follow along with the notes 😄 

I'll get back to you soon!

File: README.md [hunk 1]
```diff
@@ -2,13 +2,60 @@
...
+See below for an [Overview of Exercism][link]. If you need help navigating this documentation, see [What would you like to do?][link]. Otherwise, see the full [Table of contents][link] below.
 
+## Overview of Exercism
+
+Exercism—as an open source project—has two distinct parts.
+
+1. The Product
+2. The Curriculum
+
+The product consists of a website and a command-line client (CLI). We are currently redesigning the website, and recommend that you do not invest time and effort in submitting changes to the website at this time.
```
Comment (iHiD) [inline README.md:16]: > We are currently redesigning the website, and recommend that you do not invest time and effort in submitting changes to the website at this time.

Make change to something like: "Please check the contributing.md guidelines in exercism/website if you would like to understand how to contribute to this project" or something? I think we want to point people to a canonical place of information on this.

Comment (kytrinyx) [reply README.md:16]: I think the `CONTRIBUTING.md` of each code repository should be the setup instructions _for that repository_. I think we should have high-level contribution guidelines in a central place, such as these docs.

**Pull request closed without merging**